### PR TITLE
executor: fix a order by bug.

### PR DESCRIPTION
**Pull request by hanfei1991**
When double read requires order, we should append task to channel by order.

### CODE REVIEW

**Comment by hanfei1991** [comment]
@tiancaiamao @winoros PTAL

**Comment by tiancaiamao** [comment]
I think it's more sensible to make a goroutine, receive packet and sort the packet:

```
type Packet struct {
	id int
}

func keepOrder(in chan<- *Packet, out <-chan *Packet) {
	cache := make(map[int]*Packet)
	expectID := 0
	for pkt := range in {
		cache[pkt.id] = pkt
		expectID = drain(cache, &expectID, out, expectID)
	}
}

func drain(cache map[int]*Packet, currentID int, out <-chan *Packet) int {
	for {
		pkt, ok := cache[currentID]
		if ok {
			break
		}

		delete(cache[currentID])
		out <- pkt
		currentID++
	}
	return currentID
}

func main() {
	var ch0 chan *Packet
	ch1 := make(chan *Packet)
	// This goroutine would receive from ch0, sort packet, write result to ch1.
	go keepOrder(ch0, ch1)
	<-ch1
}
```

File: executor/new_distsql.go [hunk 1]
```diff
@@ -300,10 +300,52 @@ type IndexLookUpExecutor struct {
...
+	keepOrder  bool
+	taskCh     chan *lookupTableTask
+	curID      int
+	waitChLock sync.Mutex
+	waitChMap  map[int]chan struct{}
+}
+
+func (c *resultChannel) fetchTask() *lookupTableTask {
+	task := <-c.taskCh
+	if task != nil {
```
Comment (coocood) [inline executor/new_distsql.go]: Why check task is not nil and do nothing?

Comment (hanfei1991) [reply executor/new_distsql.go]: Redundant code..

**Comment by coocood** [comment]
I think a better way is to let the `indexWorker` send the task to both `workCh` and `resultCh`.
In Next, wait the task done.

**Comment by coocood** [comment]
Well, `getRow` already waits for the task to be done, nothing else to do, `indexWorker` just send the task to resultCh, and `tableWorker` don't, that's all.

**Comment by hanfei1991** [comment]
@coocood before @tiancaiamao 's refactor, we do it by this way. Which way is better ? Do we need discuss ? @shenli

**Pull request closed without merging**